src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1461,12 +1461,16 @@ trait Namers extends MethodSynthesis {
         // Note that matching MethodType of NullaryMethodType must be nilary not nelary.
         def overriddenNilary(sym: Symbol) = sym.info.isInstanceOf[MethodType]
         if (overridden != NoSymbol && vparamSymss.isEmpty && overridden.alternatives.exists(overriddenNilary)) {
-          if (settings.warnNullaryOverride)
-            context.unit.toCheck += {() =>
-              def javaDetermined(sym: Symbol) = sym.isJavaDefined || isUniversalMember(sym)
-              if (!meth.overrides.exists(javaDetermined))
-                context.warning(ddef.pos, "nullary method assumes an empty parameter list from the overridden definition", WarningCategory.LintNullaryOverride)
+          def notExempt() = !meth.overrides.exists(sym => sym.isJavaDefined || isUniversalMember(sym))
+          val msg = "nullary method assumes an empty parameter list from the overridden definition"
+          val toCheck = if (currentRun.isScala3) () => {
+            if (notExempt()) {
+              ErrorUtils.issueNormalTypeError(ddef, msg)
+              ddef.tpt.defineType(ErrorType): Unit
             }
+          } else () =>
+            if (notExempt()) context.warning(ddef.pos, msg, WarningCategory.LintNullaryOverride)
```

Comment:
Should use `WarningCategory.Other` as it's no longer behind an `-Xlint` flag (and remove the `LintNullaryOverride` object)

Reply:
Having the warning category already coded is a "sunk cost", right?

Reply:
Yeah, why throw the preciseness away?

Reply:
Keeping.  Un-resolve (or re-comment on the updated code) if you think it really should go.

Reply:
It should change to `OtherNullaryOverride` then, as it's no longer under the `-Xlint` flag. I think the `Lint*` categories should stay consistent.